src/core/solver.py

```
@@ -108,67 +123,97 @@ class ScoringStrategy(ABC):
-                round(p.scores.get(col, 0.0) * config.SCALE_FACTOR)
-                for p in participants
-            ]
-            # Ensure tiny positive weights are not rounded to 0
-            if weight > 0:
-                weight_m = max(1, round(weight * 100))
-            else:
-                weight_m = round(weight * 100)
-            vectors.append((col, scores, weight_m))
-        return vectors
+            raw_scores = [p.scores.get(col, 0.0) for p in participants]
+            scaled_raw = [int(round(s * config.SCALE_FACTOR)) for s in raw_scores]
```

Ruff (0.15.12)

[warning] Value being cast to `int` is already an integer

Remove unnecessary `int` call

(RUF046)